command/exclude.go

```diff
@@ -20,7 +20,7 @@ func createExcludesFromWildcard(inputExcludes []string) ([]*regexp.Regexp, error
 	for _, input := range inputExcludes {
 		if input != "" {
 			regexVersion := wildCardToRegexp(input)
-			regexpCompiled, err := regexp.Compile(regexVersion)
```

Comment:
@Kucukaslan thank you for debugging this and creating the PR. I have two cosmetic comments: 
1. Regex pattern is built in `wildCardToRegexp` function. Prefixing the `patternRegex` in that function with `(?s)` will be more convenient. We can also drop a comment about why we are using this flag.

2. `regexp.Compile("(?s)", pattern)` function calls can be encapsulated in its own method where we prepend `(?s)` with a an explanation. In this way the regex flag `(?s)`  will not be visible in multiple places in the code. 

What do you think?

Reply:
Thanks for review.

As a first note, I assume these two options are alternative options. So my further comments will be about which one is preferable and why.

### On option 1
Although the `regexp` that `wildCardToRegexp` prepared does not work accurately with new line character, I'm rather hesitated to add `(?s)` in that method. Because the `(?s)` is, at least technically, a flag and my concern is that would specifying flags be an expected behaviour of `wildCardToRegexp` method (or adding flags is the responsibility of the caller of that method)?

### On option 2
I'm not sure what kind of function do you mean.
a) If you mean a function that only takes pattern and returns the new string with `(?s)` it's doable and I'm fine with it.
b) But if you mean to replace whole `regexp.Compile("(?s)", pattern)` then I've a few concerns:
1. In the tests we use its other variant `regexp.MustCompile` which panics instead of returning error. Then we should either write seperate method for it too[^update], or same method should take another parameter "isMustCompile".
2. If we use the same wrapper method for both Compile and MustCompile the surrounding method[^sur] still needs to receive the error so it should return both the regex end the error even tough MustCompile won't return an error.

My preference would be Option 2.a if we want to handle both Compile and  MustCompile together.

[^sur]: https://github.com/peak/s5cmd/blob/acb97e9fa4d186ee47307f011fbe03f90b3e2a57/storage/url/url.go#L202-L226
[^update]: Or do you think changing wildCardToRegexp method will already fix this.

Reply:
I've extracted the regular expression preperation to three util functions: WildCardToRegexp, MatchFromStartToEnd and AddNewLineFlag.

Since the exclude.go was in `command` package but the `setPrefixAndFilter` was in `storage/url` they must be exported.
Neither of the packages seemed a correct place, so I placed them to `strutil` package. It might be better to rename `strutil` to, more generic name, `util`.


'setPrefixAndFilter' preferred `*?` over `*` which was preferred by `exclude.go`. As far as I understood, they won't affect the results since we always use full text match. But it may have some effect in the speed.